src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -1819,7 +1819,9 @@ private void PassivateCompleted(EntityId entityId)
 
         private void DeliverMessage(string entityId, object msg, IActorRef snd)
         {
-            var payload = _extractor.EntityMessage(msg)!; // payload can't be null unless dev really screwed up
+            var payload = _extractor.EntityMessage(msg); // payload can't be null unless dev really screwed up
+            if (payload is null)
+                throw new ArgumentNullException($"IMessageExtractor.EntityMessage() should never return null. Message extractor: {_extractor.GetType()}");
```

Comment:
"IMessageExtractor.EntityMessage() should never return null" message is much better than an obscure null reference exception being thrown later on in this method when we try to access the payload variable.